test/CloudNative.CloudEvents.UnitTests/Http/HttpUtilitiesTest.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright 2021 Cloud Native Foundation. 
+// Licensed under the Apache 2.0 license.
+// See LICENSE file in the project root for full license information.
+
+using System;
+using Xunit;
+
+namespace CloudNative.CloudEvents.Http.UnitTests
+{
+    public class HttpUtilitiesTest
+    {
+        [Theory]
+        [InlineData("simple", "simple")]
+        [InlineData("Euro \u20AC \U0001F600", "Euro%20%E2%82%AC%20%F0%9F%98%80")]
+        [InlineData("space encoded", "space%20encoded")]
+        [InlineData("percent%encoded", "percent%25encoded")]
+        [InlineData("quote\"encoded", "quote%22encoded")]
+        [InlineData("caf\u00e9", "caf%C3%A9")]
+        public void RoundTripHeaderValue(string original, string encoded)
```

Comment:
Would it be worth having one-way Encode/Decode tests as well?

Reply:
I started with those, but then found there was just a lot of duplication.

While we could extract the test cases into TheoryData and have two tests that both use the same TheoryData member, it felt simpler to just combine them.